src/include/processor/operator/persistent/node_batch_insert_error_handler.h

```diff
@@ -20,21 +20,6 @@ struct IndexBuilderError {
     std::optional<WarningSourceData> warningData;
 };
 
-struct IndexBuilderCachedError {
-    IndexBuilderCachedError(std::string message, std::optional<WarningSourceData> warningData);
-    IndexBuilderCachedError() = default;
-
-    template<typename T>
-    static IndexBuilderCachedError constructFrom(IndexBuilderError<T> error) {
-        return IndexBuilderCachedError{std::move(error.message), std::move(error.warningData)};
-    }
-
-    std::string message;
-
-    // CSV Reader data
-    std::optional<WarningSourceData> warningData;
-};
-
 class NodeBatchInsertErrorHandler {
```

Comment:
Common functionality for this (caching + flushing errors) was moved to `BatchInsertErrorHandler`, the only additional responsibility this class has is to delete the erroneous rows